tests/global_time_series/test_coupled_global.py

```diff
@@ -0,0 +1,355 @@
+import unittest
+from typing import Any, Dict, List
+
+from zppy.templates.coupled_global import (
+    Metric,
+    Parameters,
+    Variable,
+    VariableGroup,
+    construct_generic_variables,
+    get_data_dir,
+    get_exps,
+    get_region,
+    get_variable_groups,
+    get_vars_original,
+    get_ylim,
+    land_csv_row_to_var,
+    param_get_list,
+)
+
+
+# Helper function
+def get_var_names(vars: List[Variable]):
+    return list(map(lambda v: v.variable_name, vars))
+
+
+# Run this test suite in the environment the global_time_series task runs in.
+# I.e., whatever `environment_commands` is set to for `[global_time_series]`
+# NOT the zppy dev environment.
+# Run: python -u -m unittest tests/global_time_series/test_coupled_global.py
```

Comment:
These unit tests are specific to `coupled_global.py`. I tried to add as many unit tests as I could, but the many functions that rely on IO (notably reading data or plotting data) can't be well tested outside of integration testing/the "min-case" tests.

Importantly, these unit tests can't be run like the other unit tests. They need to be run from the environment that `zppy` is running the `global_time_series` task. That is, these tests likely need to be run from E3SM Unified rather than the `zppy` dev environment. To facilitate testing I also just moved the contents of `readTS.py` into `coupled_global.py`.

(Note: this testing setup again highlights how we need to be wary of scope creep in `zppy`. Here we're really testing the _de facto_ Global Time Series package, not so much `zppy` itself. See #398)